### PR TITLE
rados: add Create() function to create new objects

### PR DESCRIPTION
**Pull request by nixpanic**
Ceph-CSI creates new objects without initial data. This is a functionality that is missing in go-ceph at the moment.

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code

### CODE REVIEW

File: rados/ioctx.go [hunk 1]
```diff
@@ -93,6 +100,23 @@ func (ioctx *IOContext) SetNamespace(namespace string) {
...
+//
+// Implements:
+//  void rados_write_op_create(rados_write_op_t write_op, int exclusive,
+//                             const char* category)
+func (ioctx *IOContext) Create(oid string, exclusive RadosCreateOption) error {
+	c_oid := C.CString(oid)
+	defer C.free(unsafe.Pointer(c_oid))
+
+	op := C.rados_create_write_op()
+	C.rados_write_op_create(op, C.int(exclusive), nil)
```
Comment (phlogistonjohn) [inline rados/ioctx.go]: Should we be exposing these op functions directly? I usually consider "bundling"  a minor code smell, so I'd like to understand the rationale. Note that I'm ok with bundling when building conveniences in the library but I generally want this lib to not limit the user to the convenience functions.

Comment (phlogistonjohn) [reply rados/ioctx.go]: That said, feel free to tell me we can easily unbundle later and not worry if you feel that's the case. :-)

Comment (nixpanic) [reply rados/ioctx.go]: The existing `rados_create()` function is for connecting to a RADOS cluster. Calling `Create()` on the ioctx object is similar to calling `ioctx.Delete()`. There will not be a conflict in case we want to call `rados_create()` for a RADOS cluster object (`rados.Conn`).

TLDR; IMHO we can just add `ioctx.Create()` without running into conflicts.

Comment (phlogistonjohn) [reply rados/ioctx.go]: I'm a bit confused. I was referring to the "op object" and the other op related api functions. The C api appears to work like a set of discrete operations that you add to the op object to be executed with rados_write_op_operate.
What I was suggesting is that we expose the rados_create_write_op, rados_write_op_create, rados_write_op_operate, rados_release_write_op as seperate functions in go-ceph.
I am OK with having "macro" functions that build on top of those calls for convenience and was hoping to discuss the pros/cons of exposing the low level functions now vs. just doing a higher-level function like you have here and "unbundling" things later.
I do not think I was confusing this function with rados_create. :-)

Comment (phlogistonjohn) [reply rados/ioctx.go]: FWIW the Omap related functions already make use of the *_op functions in a "bundled" form. This weakens my resistance to adding another "bundled" function. I do still think we want to expose the low-level functions eventually.

Comment (nixpanic) [reply rados/ioctx.go]: Oh, I was not worried that you were confusing this with `rados_create()` :smile:  It is just that if we want to add a `Create()` function that calls `rados_create()`, there is no problem.

I agree that we probably want to add a `WriteOperation` object and accompanying operations.

**Comment by nixpanic** [comment]
Add a special error for EEXISTS, and included it in the test case.

Please review again and merge if acceptable. Thanks!

File: rados/rados.go [hunk 2]
```diff
@@ -34,6 +34,8 @@ var (
 	RadosErrorNotFound = RadosError(-C.ENOENT)
 	// RadosErrorPermissionDenied indicates a permissions issue.
 	RadosErrorPermissionDenied = RadosError(-C.EPERM)
+	// RadosErrorObjectExists indicates that an exclusive object creation failed.
+	RadosErrorObjectExist = RadosError(-C.EEXIST)
```
[the remark this answers is not in the document]
Comment (nixpanic) [reply rados/rados.go]: Ok, I'll do that, and will add a commit to rename the existing errors (+deprecate the old names).

**Comment by phlogistonjohn** [comment]
Sorry I goofed up and suggested "XXXErr" but the convention is actually "ErrXXX".  https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#error-naming

Mea culpa.
